### PR TITLE
Init structure for imports

### LINKED ISSUE #74: Imports are inconsistent
## Description

The imports are currently not consistently structured in the respective `__init__.py` files.
Subpackages are not always imported.

### PR DESCRIPTION
**Pull request by mberz**
### Which issue(s) are closed by this pull request?

Closes #74 

### Changes proposed in this pull request:
Init structure for consistent imports of respective subpackages and submodules:

- Data/container classes are top level imports (i.e. `pyfar.Signal`, `pyfar.Coordinates`)
- Sub-packages are imported into the main package and handled equivalent to sub-modules.
- Sub-packages import functions from their respective sub-modules for convenience and ease of use.
- Sub-modules in sub-packages with rare/no direct user interaction are either imported as sub-sub-modules or kept private.

As a result, imports such as `from pyfar import Signal` in modules/sub-modules need to be avoided as they might lead to cyclic import issues.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -1,24 +1,24 @@
-import pyfar
+# import pyfar
 
 
-def test_copy(sphericalvoronoi, timedata, frequencydata):
```
Comment (f-brinkmann) [inline tests/test_utils.py:4]: How do we go about this?
- rename this to `test_copy`
- move the test into the class tests, e.g., `test_coordinates::test_copy()`

**Review by f-brinkmann — CHANGES_REQUESTED** [review]

File: tests/test_imports.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        TimeData,
+        FrequencyData,
+        Coordinates,
+        Orientations,
+        Filter,
+        FilterSOS,
+        FilterFIR,
+        FilterIIR
+    ]
+    return __all__
```
Comment (sikersten) [inline tests/test_imports.py:38]: What does this do?
```suggestion
    assert __all__
```

File: pyfar/dsp/__init__.py [hunk 3]
```diff
@@ -1,11 +1,55 @@
...
-    phase, group_delay, wrap_to_2pi, nextpow2, spectrogram,
-    regularized_spectrum_inversion)
+    phase,
+    group_delay,
+    wrap_to_2pi,
+    spectrogram,
+    regularized_spectrum_inversion
+)
+
+from .filter import (
```
[the remark this answers is not in the document]
Comment (sikersten) [reply pyfar/dsp/__init__.py:16]: I'd also suggest to keep it simpler, importing the filter functionalities either via `pyfar.dsp.filter` or `pyfar.filter`. Seems to be more consistent with `pyfar.dsp.fft`

File: tests/test_imports.py (hunk 2 above)
Comment (mberz) [reply tests/test_imports.py:38]: simply avoids flake8 imported but not used warnings

File: pyfar/dsp/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mberz) [reply pyfar/dsp/__init__.py]: agreed :+1:

**Comment by mberz** [comment]
Fabian, whats up there? your suggestions are often removing and adding the same line ;D

**Comment by mberz** [comment]
Fabian, whats up there? your suggestions are often removing and adding the same line ;D

File: tests/test_utils.py (hunk 1 above)
Comment (mberz) [reply tests/test_utils.py]: renamed to test copy for now

**Comment by f-brinkmann** [comment]
> 
> 
> Fabian, whats up there? your suggestions are often removing and adding the same line ;D

It was the quickest way to suggest adding a new (second) line - we have to import pyfar.signals, which was introduced after you did the work in this Pull :)

**Review by sikersten — APPROVED** [review]

**Pull request merged**